lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -5615,6 +5633,11 @@ namespace Js
                 head->length = offset + next->length;
                 head->CheckLengthvsSize();
                 pArr->head = head;
+
+                if (oldHead != head && isInlineSegment)
```

Comment:
>if (oldHead != head && isInlineSegment) [](start = 16, length = 39)

I think if we reverse the && to have isInlineSegment first compiler can do a test for bool and then a pointer comparison which will be faster